File: practica1.py
# -*- coding: utf-8 -*-
"""
Created on Thu Feb 13 22:38:36 2025

@author: Fabricio
"""

import math

class Punto:
    def __init__(self, x: float, y: float):
        self.x = x
        self.y = y

    def coord_cartesianas(self):
        return (self.x, self.y)

    def coord_polares(self):
        r = math.sqrt(self.x**2 + self.y**2)  
        theta = math.atan2(self.y, self.x)  
        return (r, math.degrees(theta))  

    def __str__(self):  
        return f"Punto({self.x}, {self.y})"

p = Punto(5.6, 4.4)
print(p) 
print("Coordenadas Cartesianas:", p.coord_cartesianas())  
print("Coordenadas Polares:", p.coord_polares())  

File: Practica-1/src/main/java/com/mycompany/practica1/Practica1.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Project/Maven2/JavaApp/src/main/java/${packagePath}/${mainClassName}.java to edit this template
 */

package com.mycompany.practica1;

/**
 *
 * @author Fabricio
 */
public class Practica1 {

    public static void main(String[] args) {
        Punto p = new Punto(3, 4);
        System.out.println(p); 
        System.out.println("Coordenadas Cartesianas: " + p.coordCartesianas());
        System.out.println("Coordenadas Polares: " + p.coordPolares()); 
        
        Circulo c = new Circulo(p, 5f);
        System.out.println(c.toString());
        c.dibujaCirculo();
    }
}
